vdr/ambassador.go

```diff
@@ -358,8 +325,10 @@ func verifyDocumentEntryID(owner did.DID, entryID ssi.URI, knownIDs map[string]b
 	return nil
 }
 
-func isUpdate(transaction dag.SubscriberTransaction) bool {
-	return !transaction.TimelineID().Empty()
+func (n ambassador) isUpdate(doc did.Document) bool {
+	_, _, err := n.didStore.Resolve(doc.ID, nil)
```

Comment:
Shouldn't you check the resolved document as well? Because now another error (e.g. a bug in storage or when it's corrupted) causes the document to be seen as new?

Reply:
ahh yes